src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -0,0 +1,266 @@
+
+        internal readonly MsQuicNativeMethods.ListenerOpenDelegate _listenerOpenDelegate;
+        internal readonly MsQuicNativeMethods.ListenerCloseDelegate _listenerCloseDelegate;
+        internal readonly MsQuicNativeMethods.ListenerStartDelegate _listenerStartDelegate;
+        internal readonly MsQuicNativeMethods.ListenerStopDelegate _listenerStopDelegate;
+
+        internal readonly MsQuicNativeMethods.ConnectionOpenDelegate _connectionOpenDelegate;
+        internal readonly MsQuicNativeMethods.ConnectionCloseDelegate _connectionCloseDelegate;
+        internal readonly MsQuicNativeMethods.ConnectionShutdownDelegate _connectionShutdownDelegate;
+        internal readonly MsQuicNativeMethods.ConnectionStartDelegate _connectionStartDelegate;
+
+        internal readonly MsQuicNativeMethods.StreamOpenDelegate _streamOpenDelegate;
+        internal readonly MsQuicNativeMethods.StreamCloseDelegate _streamCloseDelegate;
+        internal readonly MsQuicNativeMethods.StreamStartDelegate _streamStartDelegate;
+        internal readonly MsQuicNativeMethods.StreamShutdownDelegate _streamShutdownDelegate;
+        internal readonly MsQuicNativeMethods.StreamSendDelegate _streamSendDelegate;
+        internal readonly MsQuicNativeMethods.StreamReceiveCompleteDelegate _streamReceiveCompleteDelegate;
+        internal readonly MsQuicNativeMethods.StreamReceiveSetEnabledDelegate _streamReceiveSetEnabledDelegate;
+
+        internal readonly MsQuicNativeMethods.SetContextDelegate _setContextDelegate;
+        internal readonly MsQuicNativeMethods.GetContextDelegate _getContextDelegate;
+        internal readonly MsQuicNativeMethods.SetCallbackHandlerDelegate _setCallbackHandlerDelegate;
+
+        internal readonly MsQuicNativeMethods.SetParamDelegate SetParamDelegate;
+        internal readonly MsQuicNativeMethods.GetParamDelegate GetParamDelegate;
+
+        internal unsafe uint UnsafeSetParam(
+            IntPtr Handle,
+            uint Level,
+            uint Param,
+            MsQuicNativeMethods.QuicBuffer Buffer)
+        {
+            return SetParamDelegate(
+                Handle,
+                Level,
+                Param,
+                Buffer.Length,
+                Buffer.Buffer);
+        }
+
+        internal unsafe uint UnsafeGetParam(
+            IntPtr Handle,
+            uint Level,
+            uint Param,
+            ref MsQuicNativeMethods.QuicBuffer Buffer)
+        {
+            uint bufferLength = Buffer.Length;
+            byte* buf = Buffer.Buffer;
+            return GetParamDelegate(
+                Handle,
+                Level,
+                Param,
+                &bufferLength,
+                buf);
+        }
+
+        public async ValueTask<MsQuicSecurityConfig> CreateSecurityConfig(X509Certificate certificate)
+        {
+            MsQuicSecurityConfig secConfig = null;
+            var tcs = new TaskCompletionSource<object>();
```

Comment:
You should pass TaskCreationOptions.RunContinuationsAsynchronously here, otherwise the continuation will run on the msquic thread.